src/Bicep.Core/Text/TextRange.cs

```diff
@@ -0,0 +1,64 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System.Diagnostics.CodeAnalysis;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+
+namespace Bicep.Core.Text
+{
+    [JsonConverter(typeof(SourceCodeRangeConverter))]
+    public readonly record struct TextRange(TextPosition Start, TextPosition End)
+    {
+        public TextRange(int startLine, int startColumn, int endLine, int endColumn)
+            : this(new TextPosition(startLine, startColumn), new TextPosition(endLine, endColumn))
+        {
+        }
+
+        public static TextRange? TryParse(string? value)
```

Comment:
I won't block on this, since it's pre-existing code, but it feels odd to have this logic here; feels more like this is related to the specific format used for source code bundling, rather than an innate property of a `TextRange`.

Reply:
Yeah agree. I can move this and the custom JSON type converter to the source link namespace in my next PR.